src/_findIndex.ts

```diff
@@ -0,0 +1,51 @@
+import * as _ from 'lodash';
+import { forEach } from 'lodash';
+
+const users = [
+    { user: 'barney', active: false },
+    { user: 'fred', active: false },
+    { user: 'pebbles', active: true },
+];
+
+// // console.log(
+//     _.findIndex(users, function (o) {
+//         return o.user == 'barney';
+//     })
+// );
+
+const _findIndex = <T>(array: T[], searchCondition?: T | any) => {
```

Comment:
```suggestion
const _findIndex = <T>(array: T[], searchCondition?: Partial<T> | any) => {
```
cái searchCondition kia không nhất thiết phải truyền cả object T vào, mà chỉ cần 1 số field của T -> nên dùng `Partial`

Reply:
Thay `any` thành `((item: T) => boolean)`

Reply:
Nếu như làm giống hệt Lodash thì searchCondition có thể nhận vào 1 array 2 phần tử, phần tử đầu là path, phần tử 2 là giá trị. Hoặc nhận vào 1 string.
`searchCondition: Partial<T> | ((item: T) => boolean) | [string, any] | string`